js/storage.js

```diff
@@ -1196,7 +1198,8 @@ Storage.importTeam = function (buffer, teams) {
 			if (line.substr(0, 14) === 'Hidden Power [') {
 				var hptype = line.substr(14, line.length - 15);
 				line = 'Hidden Power ' + hptype;
-				if (!curSet.ivs && window.BattleTypeChart && window.BattleTypeChart[hptype]) {
+				var shouldChangeIVs = !curSet.ivs && (!format.startsWith('gen7') || (curSet.level && curSet.level !== 100));
+				if (shouldChangeIVs && window.BattleTypeChart && window.BattleTypeChart[hptype]) {
```

Comment:
I am thinking ``!curSet.ivs || !canHyperTrain`` would lead to a better explanation of why this check is needed.

Reply:
... I guess Zarel's comment makes this point moot.